[WIP] Add Google home alarms sensor

eliseomartelli opened the pull request:
## Description:
Added a sensor to retrieve the next Google Home alarm or timer. 

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** home-assistant/home-assistant.io#7590

## Example entry for `configuration.yaml` (if applicable):
```yaml
sensor:
  - platform: google_home_alarm
    host: 192.168.0.xxx
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New or updated dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [x] New files were added to `.coveragerc`.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

eliseomartelli committed:
started working on google home alarm sensor

eliseomartelli committed:
Retrival of alarm

eliseomartelli committed:
Working state reached

eliseomartelli committed:
requirements added

eliseomartelli committed:
now the state should show the next alarm.

eliseomartelli committed:
Added icon

eliseomartelli committed:
code cosmetic changes

eliseomartelli committed:
linting

eliseomartelli committed:
small fix for indentation

eliseomartelli commented:
Added the documentation to home-assistant.io 
https://github.com/home-assistant/home-assistant.io/pull/7590

balloob commented on homeassistant/components/sensor/google_home_alarm.py:
```diff
@@ -0,0 +1,146 @@
+    vol.Optional(CONF_SHOW_TIMERS, default=True): cv.boolean,
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=['alarm', 'timer']):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+    vol.Optional(CONF_DISPLAY_OPTIONS, default='time'): vol.In(DISPLAY_TYPES)
```
display formats are a frontend concern. The backend should only deal with iso formatted timestamps

eliseomartelli replied:
will address that as soon as I get home!

eliseomartelli replied:
pushed the code now 👍

eliseomartelli replied:
The Travis build passed now

eliseomartelli committed:
now the state shows an ISO time format

eliseomartelli committed:
removed unused import

Danielhiversen commented:
Should this be a component with the google home device_tracker?

eliseomartelli commented:
> Should this be a component with the google home device_tracker?

So a "Google Home" component with a list of hosts? Seems ok... Only problem I could think of is the confusion with "Google Assistant"

frenck commented on homeassistant/components/sensor/google_home_alarm.py:
```diff
@@ -0,0 +1,103 @@
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=['alarm', 'timer']):
```
There are 2 monitored conditions.... is it really needed to make a configuration option out of this?

Also with this in mind: home-assistant/architecture#100

I would like to opt to remove this option and just expose both.

eliseomartelli replied:
Something like my latest commit?

eliseomartelli committed:
removed monitored_conditions

eliseomartelli commented:
@frenck removed the monitored condition so the user doesn't have to explicitly pick what to track.

eliseomartelli committed:
added to coveragerc

eliseomartelli committed:
set device class to timestamp

balloob commented:
there should be a google home component and it should have the same spelling as the google home device tracker. Configuration standardized under component.

eliseomartelli commented:
@balloob Perfect!

michaelarnauts commented:
It seems that the `glocalapi` project has been moved to `googledevices`.

https://pypi.org/project/ghlocalapi/ -> https://pypi.org/project/googledevices/

emontnemery committed:
Update device registry of MQTT binary_sensor

emontnemery committed:
Update device registry of MQTT climate

ludeeus committed:
Add component for googlehome

ludeeus committed:
Add missing name in CODEOWNERS

ludeeus committed:
Linting issues

ludeeus committed:
googledevices version bump

ludeeus committed:
Use NAME from component instead of DOMAIN

ludeeus committed:
Cleaner handling of accepted devices in for loop

ludeeus committed:
Fixes one linting issue

ludeeus committed:
Validate device_types

ludeeus committed:
Fixes one linting issue

ludeeus committed:
Fixes linting issue

ludeeus committed:
Revert 0abb642 and import DOMAIN as GOOGLEHOME_DOMAIN

ludeeus committed:
Return false if discovery_info is None

ludeeus committed:
Combine if's in for loop

ludeeus committed:
Use async_load_platfrom

ludeeus committed:
Fix line length

ludeeus committed:
Add error message to user

ludeeus committed:
Shorter log message

ludeeus committed:
error -> warning, remove period

dmulcahey committed:
Make imports relative in ZHA component (#20020)

* make imports relative

* remove cyclic import

balloob committed:
Switch locative to use the webhook component (#20043)

* Switch locative to use the webhook component

* Lint

* Remove dead test code

* Use voluptuous to validate the webhook schema

* Validate test mode schema as well

* Lint

* Remove allow_extra

* Return web.Response correctly

* #20043: Remove superfluous dict in WEBHOOK_SCHEMA validation

emontnemery committed:
Log exceptions thrown by MQTT message callbacks (#19977)

* Log exceptions thrown by MQTT message callbacks

* Fix tests

* Correct method for skipping wrapper in traceback

* Lint

* Simplify traceback print

* Add test

* Move wrapper to common helper function

* Typing

* Lint

fabaff committed:
Upgrade aiohttp to 3.5.3 (#19957)

* Upgrade aiohttp to 3.5.3

* Upgrade aiohttp to 3.5.4

* Remove test for webhook component from camera.push

* Lint

balloob committed:
Strip login username in backend (#20150)

* Add modern mode to HA auth provider that strips usernames

* Add tests for async_get_or_create_credentials

* Fix test

damarco committed:
Bump pynuki to 1.3.2 (#20173)

syssi committed:
Add Xiaomi Airpurifier Pro V7 support (#20093)

* Add Xiaomi Air Purifier Pro V7 support

* Reorder the model list

* Improve the list of supported attributes/properties

* Fix lint

damarco committed:
Add support for deconz radios to zha component (#20167)

* Add support for deconz radios

* Update check_zigpy_connection()

balloob committed:
Updated frontend to 20190116.0

balloob committed:
Bumped version to 0.86.0b0

boralyl committed:
Remove color call to set lights to black. (#20176)

Calling clear all is enough to turn off the light.  Calling the color
command makes the light no longer function until clear all is called
again.  The component calls clear all beforing turning it on which is
why it works through home assistant.  However if you try to control the
light via the hyperion app or through kodi after it has been turned off
via home assistant it will not function until you call clear all again.

balloob committed:
Sensibo to use HA operation modes (#20180)

balloob committed:
Sensibo to use HA operation modes (#20180)

rohankapoorcom committed:
Embed zoneminder platforms into component (#20182)

balloob committed:
Distribute reconnect (#20181)

balloob committed:
Distribute reconnect (#20181)

balloob committed:
Add command to refresh auth (#20183)

balloob committed:
Add command to refresh auth (#20183)

rohankapoorcom committed:
Add a Zoneminder availability sensor (#20184)

* Embed zoneminder platforms into component

* Add a binary sensor for ZoneMinder availability

* Lint

* Add missing docstrings

emontnemery committed:
Add JSON attribute topic to MQTT cover (#20190)

* Add JSON attribute topic to MQTT cover

* Lint

emontnemery committed:
Add JSON attribute topic to MQTT light (#20191)

emontnemery committed:
Add JSON attribute topic to MQTT switch (#20192)

emontnemery committed:
Log exceptions thrown by signal callbacks (#20015)

* Log exceptions thrown by signal callbacks

* Fix unsub

* Simplify traceback print

* Typing

* Add test

* lint

* Review comments

* Rework MQTT test case

* Fix bad merge

* Fix bad merge

ehendrix23 committed:
Set ehendrix23 as owner for harmony platform (#20203)

Put myself (ehendrix23) as code owner for remote.harmony platform

zhujisheng committed:
Add platform image_processing.qrcode (#20215)

* Add platform image_processing.qrcode

* Update qrcode.py

rohankapoorcom committed:
Add verify_ssl to restful_command and switch.rest (#20199) (#20207)

amelchio committed:
Improve Sonos discovery (#20196)

amelchio committed:
Improve Sonos discovery (#20196)

LouisMT committed:
Fix unused friendly name for SolarEdge sensor (#20109)

JoakimLindbom committed:
Fix for issue #19086 (#20225)

emontnemery committed:
Add JSON attribute topic to MQTT vacuum

OttoWinter committed:
Bump aioesphomeapi to 1.4.2 (#20247)

* Bump aioesphomeapi to 1.4.2

* Update requirements_all.txt

OttoWinter committed:
Bump aioesphomeapi to 1.4.2 (#20247)

* Bump aioesphomeapi to 1.4.2

* Update requirements_all.txt

gorynychzmey committed:
Use local IP to discover IGD device (#20035)

* Use local_ip from config to discover IGD device

In case of multi-homed server UPNP discovery finds IGD device on some "default" interface. WIth this modification discovery will be performed from 'local_ip'.

* Update device.py

* Changed version of async_upnp_client in requirements

* Used aysnc_upnp_client==0.14.0

* Changed requirement to async_upnp_client==0.14.0.dev0

* Changed requirement to async_upnp_client==0.14.0.dev0

* Changed requirement to async_upnp_client==0.14.0.dev0

* Fixed code style

* Fixed code style

* Changed version of async_upnp_client in requerements

* Changed version of async_upnp_client in requirements

* Regenerated requirements (new async_upnp_client)

* Regenerated requirements (new async_upnp_client)

* Changed requirement to async_upnp_client=0.14.1

* Changed requirement to async_upnp_client=0.14.1

* Updated requirements

* Updated requirements.txt

* Corrected requirements

* Corrected import of DeviceState

* Constants changed according new async_upnp_client

* Upgraded for async_upnp_client==0.14.2

Danielhiversen committed:
Tibber Pulse for homes without subscriptions (#20246)

mwegner committed:
Added Search Configuration to IMAP Sensor (#19749)

* Added Search Configuration to IMAP Sensor

The IMAP sensor currently only counts unread emails in a folder.  By exposing the IMAP search parameter, the sensor can be used to count other results:

- All emails in an inbox
- Emails sent from an address
- Emails matching a subject
- Other advanced searches, especially with vendor-specific extensions.  Gmail in particular supports X-GM-RAW, which lets you use any Gmail search directly ("emails with X label older than 14 days with", etc)

For my use case, I just wanted total emails in a folder, to show an "X/Y" counter for total/unread.  I started work on a one-off script to throw the data in, but figured I'd try to extend Home Assistant more directly, especially since this IMAP sensor correctly handles servers that push data.  This is my first Home Assistant contribution, so apologies in advance if something is out of place!  It's a pretty minimal modification.

* Added Server Response Checking

Looks like no library exception is thrown, so check for response text before parsing out results (previous code just counts spaces, so an error actually returns a state value of 4).

* IMAP Warning -> Error, Count Initializes to None

IMAP search response parsing throws an error instead of a warning.

Email count initializes as None instead 0.

Email count is untouched in case of failure to parse response (i.e. if server is temporarily down or throwing errors, or maybe due to user updating their authentication/login/etc).

Fixed line length on error so it fits under 80 characters.

* Fixed Indent on Logger Error

Sorry about the churn!  Python is pretty far from my daily-use language.  (I did run this one through pep8, at least)

alengwenus committed:
Support for relay ports for LCN light platform (#19632)

* Added relay ports to LCN lights platform

* Exchanged validation for ports with uppercase validator. Makes interfacing with pypck enums much more simple.

* Removed supported_features property as it is correctly inherited from parent

* Removed type annotations.

alengwenus committed:
Add LCN switch platform (#20267)

* Add LCN switch platform

* Added guard clause for discovery_info check and removed unnecessary parathesis

kbickar committed:
Updated sense library to 0.6.0 (#20271)

koreth committed:
Fire events for Lutron RadioRA2 keypad buttons (#20090)

* Add binary sensor for Lutron RadioRA2 keypad buttons

Allow automations to be triggered from RadioRA2 keypads by exposing
each button as a binary sensor.

* Remove binary sensor component; fire events directly instead.

* Address comments from code review

Danielhiversen committed:
Update Tibber library (#20273)

notoriousbdg committed:
Make Netatmo battery_percent icon dynamic (#20275)

scop committed:
Add type hints to helpers.condition (#20266)

amelchio committed:
Remove double logging of automation action (#20264)

amelchio committed:
Remove double logging of automation action (#20264)

alengwenus committed:
Add discovery_info check to LCN light platform (#20280)

* Added discovery_info check to LCN light platform

* Removed whitespaces

balloob committed:
Updated frontend to 20190120.0

balloob committed:
Updated frontend to 20190120.0

koreth committed:
Reduce log noise from Envisalink component (#20282)

amelchio committed:
Fix 'all' entity_id in service call extraction (#20281)

amelchio committed:
Fix 'all' entity_id in service call extraction (#20281)

amelchio committed:
Allow 'all' entity_id in service schema (#20278)

amelchio committed:
Allow 'all' entity_id in service schema (#20278)

rytilahti committed:
Print a message when reconnected after a connection failure, requirement for IQS silver (#20261)

balloob committed:
Add JSON attribute topic to MQTT vacuum (#20242)

emontnemery committed:
Add JSON attribute topic to MQTT lock (#20241)

emontnemery committed:
Add JSON attribute topic to MQTT fan (#20240)

emontnemery committed:
Add JSON attribute topic to MQTT climate (#20239)

emontnemery committed:
Add JSON attribute topic to MQTT alarm (#20238)

pnbruckner committed:
Handle non-string values in JSON renderer (#20233)

Handle the case of async_render_with_possible_json_value's value argument
being something other than a string. This can happen, e.g., when using the
SQL sensor to extract a datetime column such as last_changed and also using
its value_template to convert that datetime to another format. This was
causing a TypeError from json.loads, but async_render_with_possible_json_value
was only catching ValueError's.

pnbruckner committed:
Handle non-string values in JSON renderer (#20233)

Handle the case of async_render_with_possible_json_value's value argument
being something other than a string. This can happen, e.g., when using the
SQL sensor to extract a datetime column such as last_changed and also using
its value_template to convert that datetime to another format. This was
causing a TypeError from json.loads, but async_render_with_possible_json_value
was only catching ValueError's.

balloob committed:
Bumped version to 0.86.0b1

balloob committed:
Clean up build artifacts correctly

balloob committed:
Clean up build artifacts correctly

joncar committed:
Add EverLights light component (#19817)

* EverLights light integration. Supports single color (with color and brightness parameters) or saved pattern (with effect parameter).

* Fix pylint parameter name warning.

* Code review feedback.

* Add tests for the two helper functions of EverLights component.

* Fixes for review feedback.

* Change test style.

* Style fixes for hound.

andrewsayre committed:
Added recursive detection of functools.partial. (#20284)

foxel committed:
Added Xiaomi AirPurifier 2S profile (#20285)

Danielhiversen committed:
Update Tibber lib (#20289)

balloob committed:
Updated frontend to 20190121.0

balloob committed:
Updated frontend to 20190121.0

kellerza committed:
Align valid_entity_id with new slugify (#20231)

* slug

* ensure a dot

* fix

* schema_with_slug_keys

* lint

* test

kellerza committed:
Align valid_entity_id with new slugify (#20231)

* slug

* ensure a dot

* fix

* schema_with_slug_keys

* lint

* test

balloob committed:
Bumped version to 0.86.0b2

andrewsayre committed:
Remove trailing slash from base_url and added ability to generate webhook path. (#20295)

TD22057 committed:
Add radiotherm is_on method to return on/off (#20283)

* Added state method to return current operating state to fix #18244 for radiotherm component.

* Changed to set the is_on property when actively heating or cooling.

kellerza committed:
Config Validator: schema_with_slug_keys (#20298)

* schema_with_slug_keys

* Update config_validation.py

* Update config_validation.py

kellerza committed:
Config Validator: schema_with_slug_keys (#20298)

* schema_with_slug_keys

* Update config_validation.py

* Update config_validation.py

balloob committed:
Updated frontend to 20190121.1

balloob committed:
Updated frontend to 20190121.1

reefab committed:
Bugfix: prevent error notification when octoprint server auto detected but no configuration present. (#20303)

reefab committed:
Bugfix: prevent error notification when octoprint server auto detected but no configuration present. (#20303)

reefab committed:
Fix typo C02 to CO2 (#20306)

* Fix type C02 to CO2 and added VOC to air quality platform

* singularized volatile_organic_compound

* Remove VOC prop

* Update __init__.py

* Update __init__.py

balloob committed:
Bumped version to 0.86.0b3

kellerza committed:
Add git to the development Dockerfile (#20276)

* git_on_dev

* feedback

krygal committed:
Add device tracker support for EE Brightbox 2 router (#19611)

* Added device tracker support for EE Brightbox 2

* removed timeago dependency

* get scanner checks and improved tests

* fixed lint issues

* removed redundant timeago from test requirements

* fixed variable naming in test

* removed unecessary blank line

worm-ee committed:
HomematicIP add cover FROLL and BROLL devices (#19794)

* Add cover FROLL and BROLL devices

* Fix import

* Fix async calls

* Update cover functions and async fixes

* Update test

* Cleanup code

* Update header

* Merge imports

* Update

* Remove init

* Update coveragerc file

* Update coveragerc

awarecan committed:
Upgrade python-nest to 4.1.0 (#20313)

matkor committed:
Ecoal (esterownik.pl) static fuel boiler and pump controller (#18480) - matkor

* Starting work on ecoal boiler controller iface.

* Sending some values/states to controller.

* Basic status parsing, and simple settings.

* Platform configuration.

* Temp sensors seems be working.

* Switch from separate h/m/s to datetime.

* Vocabulary updates.

* secondary_central_heating_pump -> central_heating_pump2

* Pumps as switches.

* Optional enabling pumps via config.

* requests==2.20.1 added to REQUIREMENTS.

* Optional enabling temp sensors from configuration yaml.

* autopep8, black, pylint.

* flake8.

* pydocstyle

* All style checkers again.

* requests==2.20.1 required by  homeassistant.components.sensor.ecoal_boiler.

* Verify / set switches in update().
Code cleanup.

* script/lint + travis issues.

* Cleanup, imperative mood.

* pylint, travis.

* Updated .coveragerc.

* Using configuration consts from homeassistant.const

* typo.

* Replace global ECOAL_CONTR with hass.data[DATA_ECOAL_BOILER].
Remove requests from REQUIREMENTS.

* Killed .update()/reread_update() in Entities __init__()s.
Removed debug/comments.

* Removed debug/comments.

* script/lint fixes.

* script/gen_requirements_all.py run.

* Travis fixes.

* Configuration now validated.

* Split controller code to separate package.

* Replace in module docs with link to https://home-assistant.io .

* Correct component module path in .coveragerc.
More vals from  const.py.
Use dict[key] for required config keys.
Check if credentials are correct during component setup.
Renamed add_devices to add_entities.

* Sensor/switch depends on ecoal_boiler component.
EcoalSwitch inherits from SwitchDevice.
Killed same as default should_poll().
Remove not neede schedule_update_ha_state() calls from turn_on/off.

* lint fixes.

* Move sensors/switches configuration to component setup.

* Lint fixes.

* Invalidating ecoal iface cache instead of force read in turn_on/off().

* Fail component setup before adding any platform entities.
Kill NOTE.

* Disallow setting entity names from config file, use code defined default names.

* Rework configuration file to use monitored_conditions like in rainmachine component.

* Killed pylint exception.
Log error when connection to controller fails.

* A few fixes.

* Linted.

rolfberkenbosch committed:
Update locationsharinglib to version 3.0.11 (#20322)

* Update google_maps.py

There are known bug in locationsharinglib version 3.0.9, the developer has fixed this in 3.0.11. (https://github.com/costastf/locationsharinglib/issues/42)

* Update requirements_all.txt

rolfberkenbosch committed:
Update locationsharinglib to version 3.0.11 (#20322)

* Update google_maps.py

There are known bug in locationsharinglib version 3.0.9, the developer has fixed this in 3.0.11. (https://github.com/costastf/locationsharinglib/issues/42)

* Update requirements_all.txt

fabaff committed:
Upgrade youtube_dl to 2019.01.17 (#20318)

mitchellrj committed:
Should require the 'GATTOOL' setup extras which includes pexpect. (#20263)

* Should require the 'GATTOOL' setup extras which includes pexpect.

* Also fix skybeacon's requirement and requirements_all.

mitchellrj committed:
Should require the 'GATTOOL' setup extras which includes pexpect. (#20263)

* Should require the 'GATTOOL' setup extras which includes pexpect.

* Also fix skybeacon's requirement and requirements_all.

balloob committed:
Fix invalid entity ID in entity registry (#20328)

balloob committed:
Fix invalid entity ID in entity registry (#20328)

syssi committed:
Fix xiaomi speed attribute name clash (#20312)

syssi committed:
Fix xiaomi speed attribute name clash (#20312)

syphernl committed:
Add data/data_template/title to alert component (#17616)

* Add data/data_template/title to alert component

* Fix line length

* Fix tests

* Fix lint

* fix line length

* Fix tests, make title templatable

* Fix test

* Fix test

* Optimize data, make title templated

* Fix line length

* Add title template

* typo

* Fix tests

exxamalte committed:
Remove creation of geolocation default group (#20338)

reefab committed:
Add Octoprint custom path (#20302)

* Added custom path config option to octoprint component

* Added some debug logging

* removed debug logging for base url

* Fixed single/double quotes style

shbatm committed:
Update Requirement for PyISY Package in isy994 Component to v1.1.1 (#20349)

heytensai committed:
Fix mpd logging format string field (#20333)

* Fix format string field

* Remove str.format and let _LOGGER handle formatting

* Remove trailing period from log message

JonasPed committed:
Add Danfoss Air HRV support (#20138)

* Add support for Danfoss Air HRV systems.

* Correct lint errors after initial commit of Danfoss Air HRV support.

* A couple of lint fixes for danfoss_air.

* Refactor to comply with HA standards.

* Style fix.

* Use wildcard for danfoss_air in .coveragerc.

* Remove config example from header documentation.
Correct import name for platforms.

balloob committed:
Version bump to 0.86.0

Danielhiversen committed:
clean up of islamic_prayer_times (#20352)

update_sensors was not awaited in async_track_point_in_time()

fronzbot committed:
Hotfix for blink initialization failure. Fixes #20335 (#20351)

fronzbot committed:
Hotfix for blink initialization failure. Fixes #20335 (#20351)

Jc2k committed:
Bump homekit==0.12.2 + improve controller reliability (#20325)

Sessions were timing out and requiring a HA restart in order to restore
functionality. Network disconnects are now handled and sessions will be
automatically recovered after they fail.

balloob committed:
Merge pull request #20354 from home-assistant/rc

0.86.0

balloob committed:
Fix entity registry comments (#20357)

eliseomartelli committed:
[FIX] Time reporting incorrect in sensor.gtt (#20362)

* quick fix

* remove print statement

* fixes

* remove lambda

* added pylint disable

* should be fine now

eliseomartelli committed:
[FIX] Time reporting incorrect in sensor.gtt (#20362)

* quick fix

* remove print statement

* fixes

* remove lambda

* added pylint disable

* should be fine now

Danielhiversen committed:
Clean up concord232 (#20353)

* Clean up concord232

* concord cleanup

* clean up

* fix import

balloob committed:
Fix restore state crashing invalid entity ID (#20367)

balloob committed:
Fix restore state crashing invalid entity ID (#20367)

balloob committed:
Calling save before load would crash Lovelace storage (#20368)

balloob committed:
Calling save before load would crash Lovelace storage (#20368)

dgomes committed:
Fix error when API doesn't return a forecast. (#20365)

* add guard

* wrong logic

dgomes committed:
Fix error when API doesn't return a forecast. (#20365)

* add guard

* wrong logic

balloob committed:
Bumped version to 0.86.1

balloob committed:
Merge pull request #20369 from home-assistant/rc

0.86.1

Danielhiversen committed:
Change STATE_UNKOWN to None (#20337)

* Change STATE_UNKOWN to None

* Change STATE_UNKOWN to None

* tests

* tests

* tests

* tests

* tests

* style

* fix comments

* fix comments

* update fan test

fabaff committed:
Upgrade TwitterAPI to 2.5.9 (#20372)

Censored3 committed:
SongPal: do not crash if active_source is not (yet) available - fixes #20343 (#20344)

* SongPal: error handling if active_source can't be detected

* sonpal: Add comment to the use of getattr() for property source

* songpal: make comment single-line

zewelor committed:
Add yeelight start_flow service and ability to declare custom effects (#20107)

Danielhiversen committed:
Remove logging from tibber (#20382)

* Remove logging from tibber

* keep guard

amelchio committed:
Improve deprecation warnings (#20391)

mindigmarton committed:
Upgrade emulated_roku to 0.1.8 to fix invalid encodings, fixes #20388 (#20390)

starkillerOG committed:
Philips Hue - Remove unnessesary warning (#20394)

for white hue bulbs and bulbs from other brands like Ikea and Innr, this warning will be issued while this is not really a problem.
So just remove the warning.

starkillerOG committed:
Philips Hue - Remove unnessesary warning (#20394)

for white hue bulbs and bulbs from other brands like Ikea and Innr, this warning will be issued while this is not really a problem.
So just remove the warning.

balloob committed:
history allowed to load states with invalid entity IDs (#20399)

balloob committed:
history allowed to load states with invalid entity IDs (#20399)

fabaff committed:
Upgrade numpy to 1.16.0 (#20396)

danielperna84 committed:
Update pyhomematic to 0.1.55 (#20397)

balloob committed:
Bumped version to 0.86.2

balloob committed:
Merge pull request #20402 from home-assistant/rc

0.86.2

jvanderneutstulen committed:
Return windspeed and windgust in km/h instead of m/s. (#20340)

Darksky dev docs state (https://darksky/dev/docs):
`ca: same as si, except that windSpeed and windGust are in kilometers per
hour`

emontnemery committed:
Update device registry of MQTT Switch (#19540)

* MQTT Switch: Update device registry

* Move config_entry to constructor

* Remove duplicated code

* Fix merge error

emontnemery committed:
Cleanup if discovered mqtt light can't be added (#19740)

* Cleanup if discovered mqtt light can't be added

* No bare except

* Clear ALREADY_DISCOVERED list with helper

* Use constant instead of string literal

emontnemery committed:
Add character encoding to MQTT automation. (#20292)

amelchio committed:
Add LIFX listen port advanced configuration (#20299)

foxel committed:
Add time_throttle filter to sensor.filter (#20334)

* Added time_throttle filter

* Added time_throttle filter test

* Updated comments for time_throttle filter

jonudewux committed:
Upgrade youtube_dl to 2019.01.24 (#20408)

Jc2k committed:
Move homekit_controller entity types under homekit_controller platform (#20376)

* Move homekit_controller entity types under homekit_controller platform

* Update coveragerc as homekit_controller moved

zewelor committed:
Fix cast platform album name property (#20411)

coreGreenberet committed:
Fix minor homematicip cloud binary sensor issues (#20398)

* fix for smoke detection

* a tilted window is now considered as "open"/on

* changed comparison to enum

* line length

* insert brackets for line length and comparison

* indentation should now be ok for hound
changed api version to 0.10.4

* indentation should now be ok for hound
changed api version to 0.10.4

* updating requirement files

* satisfy lint

coreGreenberet committed:
Fix homematicip cloud alarm_arm_home (#20321)

kennedyshead committed:
Bumps aioasuswrt (#20432)

LouisMT committed:
Fix HTTP login attempts check triggering too late (#20431)

sqldiablo committed:
WeMo - Fix device discovery issues (#20446)

adrum committed:
Add alarm type workaround zwave lock Yale YRD240 (#20438)

amelchio committed:
Error handling for recorder purge (#20424)

dmulcahey committed:
ZHA component rewrite part 1  (#20456)

* rearrange files

* add init to module

* update imports

* update coveragerc

* put blank line back... git raw view be damned

fabaff committed:
Upgrade astral to 1.8 (#20459)

JonasPed committed:
Correct minor comments from PR#20138. (#20454)

dgomes committed:
Utility meter (#19718)

* initial commit

* test service calls

* lint

* float -> Decimal

* extra tests

* lint

* lint

* lint

* lint

* fix self reset

* clean

* add services

* improve service example description

* add optional paused initialization

* fix

* travis fix

* fix YEARLY

* add tests for previous bug

* address comments and suggestions from @ottowinter

* lint

* remove debug

* add discoverability capabilities

* no need for _hass

* Update homeassistant/components/sensor/utility_meter.py

Co-Authored-By: dgomes <diogogomes@gmail.com>

* Update homeassistant/components/sensor/utility_meter.py

Co-Authored-By: dgomes <diogogomes@gmail.com>

* correct comment

* improve error handling

* address @MartinHjelmare comments

* address @MartinHjelmare comments

* one patch is enought

* follow @ballob suggestion in https://github.com/home-assistant/architecture/issues/131

* fix tests

* review fixes

* major refactor

* lint

* lint

* address comments by @MartinHjelmare

* rename variable

danielsjf committed:
Add Co2signal sensor (#19204)

* Initial commit for the co2signal sensor

* Clean code

* Run script gen_requirements_all.py

* remove unintended character

* Remove redundancy

* Remove unused imports

* Code style

* Code style fixes

* Code style

* Fix comments PR

Comments by @fabaff
* Remove redundant comments and variables
* Follow the latest home-assistant guidelines

* Bump CO2Signal version

* Round API result

* Improve default latitude/longitude handling

* Improve friendly name

* Improve config handling

* Make lines shorter

* Style

* Convert default to variable

None does not pass cv.string

* Message if not inclusive

* Shorten line

* Update requirements

* Update co2signal.py

Group imports;  remove empty lines; refactor use of location_type; remove logging messages; remove unused functions; add global variables

* Update co2signal.py

Import platform schema from sensor

* Small fix

* Update co2signal.py

Remove last mentions of location_type

* Review changes

Add attribution
Formatting

* Missing whitespace

* Update co2signal.py

Fix pylint

* Update co2signal.py

Change blank lines

* Update co2signal.py

Initialise _data

dmulcahey committed:
introduce gateway (#20460)

emontnemery committed:
Update device registry of MQTT fan (#20445)

balloob committed:
Update device registry of MQTT climate (#20444)

balloob committed:
Merge pull request #20442 from emontnemery/mqtt_binary_sensor_update_device_info

Update device registry of MQTT binary_sensor

emontnemery committed:
Update device registry of MQTT sensor (#20440)

emontnemery committed:
Update device registry of MQTT alarm (#20439)

emontnemery committed:
Update device registry of MQTT cover (#20443)

* Update device registry of MQTT cover

* Move config_entry to constructor

starkillerOG committed:
Philips Hue, include debug message for color gamut (#20455)

starkillerOG committed:
Philips Hue, include debug message for color gamut (#20455)

fabaff committed:
Upgrade sqlalchemy to 1.2.16 (#20474)

fabaff committed:
Upgrade psutil to 5.5.0 (#20462)

balloob committed:
Warn for old slugs/entity ids (#20478)

* Warn for old slugs/entity ids

* add comments

* Lint

* LInt

* Lint

* Lint

balloob committed:
Warn for old slugs/entity ids (#20478)

* Warn for old slugs/entity ids

* add comments

* Lint

* LInt

* Lint

* Lint

shaan7 committed:
Include exception details in the error log (#20461)

* Include exception details in the error log

I see this error quite often in my HA logs and this will be helpful for anyone who is attempting to debug this.

* Minor change

* Remove line break

balloob committed:
Bumped version to 0.86.3

balloob committed:
Merge pull request #20485 from home-assistant/rc

0.86.3

balloob committed:
Bumped version to 0.87.0.dev0

rohankapoorcom committed:
Fix linting error (#20488)

bachya committed:
Use HASS latitude/longitude as defaults for Lyft (#20491)

rohankapoorcom committed:
Upgrade zm-py to 0.3.1 (#20489)

rohankapoorcom committed:
Split out dovado to a component and sensor platform (#20339)

* Split out dovado to a component and sensor platform

* Lint

* Address code review comments (#20339)

* Switch to using a notify platform for dovado SMS (#20339)

* Optimizing imports

* Remove return on `setup_platform`.

* Clean up unneeded constants

Jc2k committed:
Add homekit_controller tests (#20457)

* Add a test for a homekit_controller switch

* Add a test for a homekit_controller lightbulb

* Add a test for homekit_controller thermostat

* Changes from review

* Patch utcnow to known time in HK tests

* Neater fixture use per review

StevenLooman committed:
Upgrade to async-upnp-client==0.14.3 (#20502)

Danielhiversen committed:
Upgrade tibber library (#20504)

* Upgrade tibber library

* Upgrade tibber library

emontnemery committed:
Update device registry of MQTT Lock (#20501)

emontnemery committed:
Update device registry of MQTT Vacuum (#20500)

emontnemery committed:
Small cleanup of MQTT platforms (#20503)

* Move CONF_UNIQUE_ID to init

* Sort imports

* Update ordering

Danielhiversen committed:
Upgrade mill library (#20514)

hfurubotten committed:
Add nilu air_quality platform (#19674)

* Add nilu air_pollutants platform

* Code Review - validation, DRYs, rm state override, new attr
- Repeated code moved to own method.
- Removed override of state property.
- New attr for showing nilu pollution recommendations.
- More validation of stations input.
- Minor fixes and typos.

* Removed unused prop

* Check for none result from client before entity add

* Moved platform to air_quality component

* Updated outdated docstrings

* Minor changes

eliseomartelli committed:
Add Iliad Italy (Mobile Telephony Provider) Sensor (#19645)

* working state

* Attrs

* > requirements generated

* linting

* fixes

* coveragerc

* ordered imports and fixed auth

* Added Throttle

* moved throttle to decorator

* remove scan interval

* lower throttle

* moved to scan interval

* Add attribution

rohankapoorcom committed:
Load/unload geofency entities correctly between component and platform (#20497)

* Load/unload geofency entities correctly between component and platform

* Lint

* Await the coroutine directly

rohankapoorcom committed:
Load/unload locative entities correctly between component and platform (#20498)

* Embed device_tracker in locative

* Load/unload locative entities correctly between component and platform

* Await the coroutine directly

* Await the correct coroutine

rohankapoorcom committed:
Load/unload gpslogger entities correctly between component and platform (#20448)

* Embed device_tracker in gpslogger

* Load/unload gpslogger entities correctly between component and platform

* Await the coroutine directly

davidlie committed:
Use more up-to-date version of pyfoscam library (#20419)

* Change foscam python library to pyfoscam, which is more up to date and has several critical bug fixes.

* Update requirements_all.txt to match.

* Inserting automatically generated requirements.txt

chrillux committed:
Reset Brottsplatskartan incident types every day (#20117)

* Reset the incident types count every day

* Remove functionality that was never implemented

We don't need to keep track of previous incidents because it's not used
anywhere.

* Create empty dictionary with a pair of braces: {}

Jc2k committed:
Add more HomeKit controller tests (#20515)

* homekit_controller tests: automatically find entity ids in tests

Some entities use dynamic ids because of the nature of the test fakes it is
hard to predict the name of the entity that will be created. This inspects the
EntityComponent of the domain to find the freshly created entity.

* homekit_controller: Tests can now define their own Service models.

All existing tests use models as defined upstream. But upstream only defines a
few service models. This adds a generic model helper for creating test
service/characteristic models.

* homekit_controller: Add cover tests

* homekit_controller: Add lock tests

* homekit_controller: Add alarm_control_panel tests

* homekit_controller: Update light tests for color_temp.

* Revert "homekit_controller tests: automatically find entity ids in tests"

This reverts commit 506caa4c3e0814eec637145d7d6eaf2bd642e99e.

* homekit_controller: Mock entity name so entity_id is consistent.

Also remove spurious subclass overrides that are identical to parent class.

* homekit_controler: Make tests less awkward as allowed top level imports

Jc2k committed:
Homekit controller BLE groundwork (#20538)

* Define the characteristics to poll (or subscribe to) up front

* Configure characteristics immediately instead of during first poll

* Do as much cover configuration upfront as possible

* Remove test workaround as no longer needed

* Remove switch code that is already handled by HomeKitEntity

* Remove lock code already handled by HomeKitEntity

* Remove light code already handled by HomeKitEntity

* Remove alarm code already handled by HomeKitEntity

* Remove climate code already handled by HomeKitEntity

Jc2k committed:
Homekit controller BLE groundwork (part 2) (#20548)

* Only fetch values of characteristics we are tracking.

* Use callbacks on subclasses to update individual values

* Update alarm_control_panel to use update callbacks

* Update climate to use update callbacks

* Update cover to use update callbacks

* Update light to use update callbacks

* Update lock to use update callbacks

* Update switch to use update callbacks

* Remove compatibility code as all entities migrated

* pylint by name rather than code

bachya committed:
Make Ambient PWS async and cloud-push (#20332)

* Moving existing sensor file

* Initial functionality in place

* Added test for config flow

* Updated coverage and CODEOWNERS

* Linting

* Linting

* Member comments

* Hound

* Moving socket disconnect on HASS stop

* Member comments

* Removed unnecessary dispatcher call

* Config entry fix

* Added support in config flow for good accounts with no devices

* Hound

* Updated comment

* Member comments

* Stale docstrings

* Stale docstring

Kane610 committed:
Area registry (#20435)

* First draft of area registry

* Refactor based on input

* Add tests for areas
Add tests for updating device

* Updating a device shouldn't require area

* Fix Martins comment

* Require admin

* Save after deleting

* Rename read to list_areas
Fix device entry_dict
Remove area id from device when deleting area

* Fix tests

emontnemery committed:
Cleanup if discovered mqtt vacuum can't be added (#20549)

emontnemery committed:
Add COMPONENT_SCHEMA and use it in alarm_control_panel (#20224)

* Add COMPONENT_SCHEMA and use in alarm and mqtt

* Revert MQTT changes

* Lint

* Small tweak

* Add tests

* Rename COMPONENT_SCHEMA to PLATFORM_SCHEMA_BASE

* Fix tests

* Improve tests

emontnemery committed:
Add discovery update support to MQTT camera (#20529)

dmulcahey committed:
ZHA component rewrite part 3 - update helpers (#20463)

* update helpers

* address comments

* remove ieee

* cluster id as hex too

fredrike committed:
fix #20387 devices without model/protocol (#20530)

perosb committed:
Add missing switch for motion_detect (#20540)

starkillerOG committed:
Add check to validate gamut (#20518)

* color.util - Add check to validate gamut

* fix indents

* fix typo

* Add check to validate gamut

* Add tests for gamut checker

* fix test

* fix pylint issues

* fix hue light gamut tests

* add check to validate gamut

* move None check

* Move None check

* Include prompt to update bridge/bulb on error

* fix wrong commit

* fix error message

* Update light.py

emontnemery committed:
Update device registry of MQTT light (#20441)

* Update device registry of MQTT light

* Move config_entry to constructor

rohankapoorcom committed:
Move CONF_UPDATE_INTERVAL to homeassistant.const (#20526)

Jc2k committed:
Add support for HomeKit motion sensor devices (#20555)

dgomes committed:
Add an Integration sensor (#19703)

* initial version

* Tested

* set state added

* lint

* lint

* remove artifacts

* Use Decimal instead of float

* simplify

* travis lint fix

* addres comments by @ottowinter

* remove job

* better sanity check

* lower error -> warning

* hound

* fix state validation

* refactor energy -> integration

* address @MartinHjelmare comments

* new style string format

* remove async_set_state

* patching the source function

balloob committed:
Allow usernames to be case-insensitive (#20558)

* Allow usernames to be case-insensitive

* Fix typing

* FLAKE*

aerialls committed:
Add Synology SRM device tracker (#20320)

mitchellrj committed:
Treat each player as a 'device' for non-client devices. (#20074)

stealthhacker committed:
Add Recollect Waste (#20121)

* Added Recollect Waste Curbside Collection sensor for tracking next collection date and for which types of waste.

* Added missing schema attributes.

* Adding requirements and coverage entries for Recollect Waste platform.

* Added exception handling, some other fixes and suggestions from code review.

* Fixed reference to incorrect exception type.

* Updated requirements_all.txt with new version of recollect-waste.

* Added true to add_entities. Created constant for default time interval. Used different pylint exclusion comment.

* Using HA's CONF_SCAN_INTERVAL now. Unique_id is now set in @property.

* Changed parameter of timedelta from seconds to days.

* Added test run of recollect client during setup_platform. Using built in SCAN_INTERVAL now.

* Return nothing in setup_platform if there is an exception.

MatteGary committed:
New Transmission component (#19230)

* New Transmission component and interaction

First commit for New Transmission component and interaction

* Fix commit

* Fix commit

* Fix + Switch checkin

Fix according to failed build and request, first checkin for Turtle Mode Switch in Transmission, still have to figure it out why it's not working.

* Bugfixing

* Fix commit

Multiple fix

* Fix

* fix for missing config

* Update on requirements_all.txt

* Fix in requirements_all.txt

* Fix

* Fix for build

* fix

* Fix

* Fix (again)

* Fix

* Fix indentation

* Fix indentation

* Fix Throttle

* Update .coveragerc

* Fix import and coveragerc

kennedyshead committed:
Fixing the openssl issue (#20570)

Jc2k committed:
Homekit Motion Sensor Review feedback (#20568)

eliseomartelli committed:
Merge branch 'google-home-alarms' of https://github.com/eliseomartelli/home-assistant into google-home-alarms
